### PR TITLE
added docs for date types and predicates supported in elastic

### PR DESCRIPTION
**Pull request by sumannewton**
- Document all the `date` types supported in elasticsearch. Also document what is not supported as of now.
- Document all the filters/predicates pushdown to elasticsearch to help users to configure the elasticsearch mapping.

### CODE REVIEW

**Comment by sumannewton** [comment]
retest this please

**Comment by sumannewton** [comment]
test this please

**Comment by sumannewton** [comment]
@martint @aalbu @phd3 Requesting your review. Thanks.

File: docs/src/main/sphinx/connector/elasticsearch.rst [hunk 1]
```diff
@@ -225,6 +228,17 @@ property definition to the ``_meta.presto`` property of the target index mapping
         }
     }'
 
+Date types
+^^^^^^^^^^
+
+Elasticsearch supports a wide array of `date`_ formats including `built in date formats`_ and also `custom date formats`_.
+Currently the elasticsearch connector supports neither built in date formats nor custom formats. Date with `format`_ property is ignored.
```
Comment (bitsondatadev) [inline docs/src/main/sphinx/connector/elasticsearch.rst:235]:
```suggestion
Currently, the elasticsearch connector supports neither built in date formats nor custom formats. Dates with the `format`_ property are ignored.
```

File: docs/src/main/sphinx/connector/elasticsearch.rst [hunk 2]
```diff
@@ -252,6 +266,28 @@ as part of the table name, separated by a colon. For example:
 
 .. _full text query: https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-query-string-query.html#query-string-syntax
 
+Predicates push down
```
Comment (bitsondatadev) [inline docs/src/main/sphinx/connector/elasticsearch.rst:269]:
```suggestion
Predicate push down
```

**Review by bitsondatadev — APPROVED** [review]
Awesome Suman! Just a few changes but after that I approve x1000. Much needed!

File: docs/src/main/sphinx/connector/elasticsearch.rst (hunk 1 above)
Comment (sumannewton) [reply docs/src/main/sphinx/connector/elasticsearch.rst]: Done.

File: docs/src/main/sphinx/connector/elasticsearch.rst (hunk 2 above)
Comment (sumannewton) [reply docs/src/main/sphinx/connector/elasticsearch.rst]: Done

File: docs/src/main/sphinx/connector/elasticsearch.rst [hunk 3]
```diff
@@ -252,6 +266,28 @@ as part of the table name, separated by a colon. For example:
 
 .. _full text query: https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-query-string-query.html#query-string-syntax
 
+Predicate push down
+-------------------
+
+Supports predicates push down of below data types:
```
Comment (bitsondatadev) [inline docs/src/main/sphinx/connector/elasticsearch.rst:272]: Missed this: predicate push down

**Review by bitsondatadev — APPROVED** [review]
Sorry I missed one. Otherwise I approve.

File: docs/src/main/sphinx/connector/elasticsearch.rst (hunk 3 above)
Comment (sumannewton) [reply docs/src/main/sphinx/connector/elasticsearch.rst]: Done.

File: docs/src/main/sphinx/connector/elasticsearch.rst [hunk 4]
```diff
@@ -225,6 +228,17 @@ property definition to the ``_meta.presto`` property of the target index mapping
         }
     }'
 
+Date types
+^^^^^^^^^^
+
+Elasticsearch supports a wide array of `date`_ formats including `built in date formats`_ and also `custom date formats`_.
+Currently, the elasticsearch connector supports neither built in date formats nor custom formats. Dates with the `format`_ property are ignored.
```
Comment (hashhar) [inline docs/src/main/sphinx/connector/elasticsearch.rst:235]: Sounds confusing. If it supports neither then what is actually supported?

It's generally better to word these things as the supported case. i.e. `Currently the elasticsearch connector supports only xyz dates. All other date formats including built-in or custom are not supported.`?

Comment (sumannewton) [reply docs/src/main/sphinx/connector/elasticsearch.rst:235]: Oh yeah. When I re-read it, even I felt like confusing. Let me rephrase it.
`Currently, the elasticsearch connector supports only default date type. All other date formats including built-in and custom formats are not supported. Date with the `format`_ property is ignored.`

Comment (hashhar) [reply docs/src/main/sphinx/connector/elasticsearch.rst:235]: Yes it does. Thanks.

**Comment by sumannewton** [comment]
@martint @mosabua @findepi Requesting your review.

File: docs/src/main/sphinx/connector/elasticsearch.rst [hunk 5]
```diff
@@ -225,6 +228,18 @@ property definition to the ``_meta.presto`` property of the target index mapping
         }
     }'
 
+Date types
+^^^^^^^^^^
+
+Elasticsearch supports a wide array of `date`_ formats including `built-in date formats`_ and also `custom date formats`_.
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/elasticsearch.rst:234]: Wrap at 80 char width please

File: docs/src/main/sphinx/connector/elasticsearch.rst [hunk 6]
```diff
@@ -225,6 +228,18 @@ property definition to the ``_meta.presto`` property of the target index mapping
...
+Currently, the elasticsearch connector supports only default **date** type. All other date formats including `built-in date formats`_ and `custom date formats`_ are not supported.
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/elasticsearch.rst:235]: The Elasticsearch connector supports only ..

**Review by mosabua — CHANGES_REQUESTED** [review]
Just some minor input. I dont know about technical correctness.

File: docs/src/main/sphinx/connector/elasticsearch.rst (hunk 5 above)
Comment (sumannewton) [reply docs/src/main/sphinx/connector/elasticsearch.rst]: I have wrapped it here now. But the same convention hasn't been followed throughout the document.

File: docs/src/main/sphinx/connector/elasticsearch.rst (hunk 6 above)
Comment (sumannewton) [reply docs/src/main/sphinx/connector/elasticsearch.rst]: Addressed.

File: docs/src/main/sphinx/connector/elasticsearch.rst (hunk 5 above)
Comment (mosabua) [reply docs/src/main/sphinx/connector/elasticsearch.rst]: I know .. we will have to do a bulk wrap edit at some stage.. for now we have been doing piecemeal improvements so doing it for new content helps. And also works better for reviews..

File: docs/src/main/sphinx/connector/elasticsearch.rst (hunk 6 above)
Comment (mosabua) [reply docs/src/main/sphinx/connector/elasticsearch.rst]: Doesnt look like it

**Review by mosabua — CHANGES_REQUESTED** [review]
I think you must have forgotten to push the improvements based on my comments.. current content is still wrong.

File: docs/src/main/sphinx/connector/elasticsearch.rst [hunk 7]
```diff
@@ -225,6 +228,21 @@ property definition to the ``_meta.presto`` property of the target index mapping
         }
     }'
 
+Date types
+^^^^^^^^^^
+
+The Elasticsearch supports a wide array of `date`_ formats including
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/elasticsearch.rst:234]: Elasticsearch supports

File: docs/src/main/sphinx/connector/elasticsearch.rst [hunk 8]
```diff
@@ -225,6 +228,21 @@ property definition to the ``_meta.presto`` property of the target index mapping
...
+`built-in date formats`_ and also `custom date formats`_.
+The Elasticsearch connector supports only default **date** type. All other date
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/elasticsearch.rst:236]: use code highlight
```
default ``date`` type.
```

File: docs/src/main/sphinx/connector/elasticsearch.rst [hunk 9]
```diff
@@ -225,6 +228,21 @@ property definition to the ``_meta.presto`` property of the target index mapping
...
+formats including `built-in date formats`_ and `custom date formats`_ are not
+supported.
+Dates with the `format`_ property are ignored.
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/elasticsearch.rst:239]: move up to prior line

File: docs/src/main/sphinx/connector/elasticsearch.rst [hunk 10]
```diff
@@ -252,6 +270,28 @@ as part of the table name, separated by a colon. For example:
 
 .. _full text query: https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-query-string-query.html#query-string-syntax
 
+Predicate push down
+-------------------
+
+Supports predicate push down of below data types:
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/elasticsearch.rst:276]: The connector supports predicate ...

**Review by mosabua — CHANGES_REQUESTED** [review]

**Review by mosabua — APPROVED** [review]
Looks good now. Thanks!

File: docs/src/main/sphinx/connector/elasticsearch.rst [hunk 11]
```diff
@@ -225,6 +228,20 @@ property definition to the ``_meta.presto`` property of the target index mapping
         }
     }'
 
+Date types
+^^^^^^^^^^
+
+Elasticsearch supports a wide array of `date`_ formats including
+`built-in date formats`_ and also `custom date formats`_.
+The Elasticsearch connector supports only default ``date`` type. All other date
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/elasticsearch.rst:236]: only the default

Comment (sumannewton) [reply docs/src/main/sphinx/connector/elasticsearch.rst]: Done.

**Comment by sumannewton** [comment]
@martint @findepi @mosabua Addressed all review comments. Re-requesting your review.

**Comment by martint** [comment]
Can you rephrase the commit message as "Add docs for date types and predicates supported in elasticsearch", or "Document date types and supported predicates in elasticsearch" ?

**Comment by sumannewton** [comment]
> Can you rephrase the commit message as "Add docs for date types and predicates supported in elasticsearch", or "Document date types and supported predicates in elasticsearch" ?

Done.
